girder_annotation/girder_large_image_annotation/rest/annotation.py

```diff
@@ -558,6 +561,94 @@ def deleteItemAnnotations(self, item):
+                        }}},
+                    {'access.groups':
+                        {'$elemMatch': {
+                            'id': {'$in': user['groups']},
+                            'level': {'$gte': 2}
+                        }}}
+                ]
+            }}
+        ] if not user['admin'] else []
+        pipeline = [
+            {'$match': {'_id': 'none'}},
+            {'$unionWith': {
+                'coll': 'folder',
+                'pipeline': [{'$match': {'_id': ObjectId(id)}}] +
+                recursivePipeline +
+                [{'$unionWith': {
+                    'coll': 'folder',
+                    'pipeline': [{'$match': {'_id': ObjectId(id)}}]
+                }}, {'$lookup': {
+                    'from': 'item',
+                    'localField': '_id',
+                    'foreignField': 'folderId',
+                    'as': '__items'
+                }}, {'$lookup': {
+                    'from': 'annotation',
+                    'localField': '__items._id',
+                    'foreignField': 'itemId',
+                    'as': '__annotations'
+                }}, {'$unwind': '$__annotations'},
+                    {'$replaceRoot': {'newRoot': '$__annotations'}},
+                    {'$match': {'_active': {'$ne': False}}}
+                ] + accessPipeline
+            }},
+        ]
+        pipeline = pipeline + [{'$sort': {sort: sortDir}}] if sort else pipeline
+        pipeline = pipeline + [{'$skip': offset}] if offset else pipeline
+        pipeline = pipeline + [{'$limit': limit}] if limit else pipeline
+
+        return Annotation().collection.aggregate(pipeline)
+
+    @autoDescribeRoute(
+        Description('Check if there are any annotations from the items in a folder')
+        .param('id', 'The ID of the folder', required=True, paramType='path')
+        .param('recurse', 'Whether or not to recursively check '
+               'subfolders for annotations', required=False, default=True, dataType='boolean')
+        .errorResponse()
+    )
+    @access.public
+    def existFolderAnnotations(self, id, recurse):
+        annotations = self.getFolderAnnotations(id, recurse, self.getCurrentUser(), 1)
+        try:
+            next(annotations)
+            yield True
+        except StopIteration:
+            yield False
+
+    @autoDescribeRoute(
+        Description('Get the annotations from the items in a folder')
+        .param('id', 'The ID of the folder', required=True, paramType='path')
+        .param('recurse', 'Whether or not to retrieve all '
+               'annotations from subfolders', required=False, default=False, dataType='boolean')
+        .pagingParams(defaultSort='created', defaultSortDir=-1)
+        .errorResponse()
+    )
+    @access.public
+    def returnFolderAnnotations(self, id, recurse, limit, offset, sort):
+        return self.getFolderAnnotations(id, recurse, self.getCurrentUser(), limit, offset,
```

Comment:
If the returned result has a `count` method, girder will automatically add a header with the count of results for a list-like entry.  This could be done like here: https://github.com/girder/girder/blob/master/girder/utility/acl_mixin.py#L345-L351.  

A simple way to do this might be to add a `count` parameter to `getFolderAnnotations` where, if True, instead of adding the sort, skip, and limit fields to the pipeline, you'd do `pipeline += [{'$count': 'count'}]`, then this endpoint would become
```
        result = self.getFolderAnnotations(
            id, recurse, self.getCurrentUser(), limit, offset, sort[0][0], sort[0][1])

        def count():
            try:
                return next(iter(self.getFolderAnnotations(
                    id, recurse, self.getCurrentUser(), count=True)))['count']
            except StopIteration:
                # If there are no values, this won't return the count, in
                # which case it is zero.
                return 0

        result.count = count
        return result
```

Reply:
Oh that's good to know, thanks!

See 0e1adec